### PR TITLE
[WIP] Move mesos_maintenance stuff into its own file

### PR DESCRIPTION
**Pull request by solarkennedy**

### CODE REVIEW

**Comment by solarkennedy** [comment]
Same failure: 
https://travis-ci.org/Yelp/paasta/jobs/158613257#L563

I don't get it :(

File: paasta_tools/mesos_maintenance.py [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    """
+    schedules = get_maintenance_schedule().json()
+    current_time = datetime_to_nanoseconds(now())
+    ret = []
+    if 'windows' in schedules:
+        for window in schedules['windows']:
+            end = window['unavailability']['start']['nanoseconds'] + window['unavailability']['duration']['nanoseconds']
+            if end < current_time:
+                ret += [host['hostname'] for host in window['machine_ids']]
+    log.debug("Hosts past maintenance end: {0}".format(ret))
```
Comment (nhandler) [inline paasta_tools/mesos_maintenance.py]: You never return `ret`, so your test fails. Testing it locally, if you `return ret` after the `log.debug`, all of the unit tests pass.

**Comment by Rob-Johnson** [comment]
lg2m - I'd like to see the api client separated into it's own module and all this under a 'maintenance' subpackage, but I'm happy if you want to ticket me to do the work myself once you're done here; this is a good first step.

**Comment by solarkennedy** [comment]
Sorry, I'm out of steam to move this again. I've rebased and finally got the tests to pass.

**Pull request merged**